api/ocm/extensions/repositories/genericocireg/config/type.go

```diff
@@ -0,0 +1,128 @@
+package config
+
+import (
+	"strings"
+
+	"ocm.software/ocm/api/config"
+	cfgcpi "ocm.software/ocm/api/config/cpi"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+const (
+	ConfigType   = "blobLimits.ocireg.ocm" + cfgcpi.OCM_CONFIG_TYPE_SUFFIX
+	ConfigTypeV1 = ConfigType + runtime.VersionSeparator + "v1"
+)
+
+func init() {
+	cfgcpi.RegisterConfigType(cfgcpi.NewConfigType[*Config](ConfigType, usage))
+	cfgcpi.RegisterConfigType(cfgcpi.NewConfigType[*Config](ConfigTypeV1))
+}
+
+// Config describes a memory based config interface
+// for configuring blob limits for underlying OCI manifest layers.
+type Config struct {
+	runtime.ObjectVersionedType `json:",inline"`
+	// BlobLimits describe the limit setting for host:port
+	// entries. As a spcial case (for testing) it is possible
+	// to configure linits for CTF, also, by using "@"+filepath.
+	BlobLimits BlobLimits `json:"blobLimits"`
+}
+
+type BlobLimits map[string]int64
```

Comment:
I wonder. I've seen, and please correct me if I'm wrong, that this could potentially be called concurrently?

In that case, this will race.

Reply:
yes, but the config object should be configured before adding it to the config context.
Modifying config object once handled by the config context will always lead to unpredictable results
and is there not allowed.

Reply:
Okay cool. Thanks for the explanation!